/*  Copyright (C) 2025 José Rebelo

    This file is part of gadgetbridge.

    gadgetbridge is free software: you can redistribute it and/or modify
    it under the terms of the GNU Affero General Public License as published
    by the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    gadgetbridge is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU Affero General Public License for more details.

    You should have received a copy of the GNU Affero General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>. */
package nodomain.freeyourgadget.gadgetfit.entities;

import nodomain.freeyourgadget.gadgetfit.model.SleepScoreSample;
import nodomain.freeyourgadget.gadgetfit.service.btle.BLETypeConversions;

public abstract class AbstractHuamiSleepSessionSample extends AbstractTimeSample implements SleepScoreSample {
    public abstract byte[] getData();

    @Override
    public int getSleepScore() {
        return BLETypeConversions.toUnsigned(getData(), 0x16);
    }
}
